Version 0.5.3

kcf-jackson opened the pull request:
- Add integration with VS Code
- Improve error tracking
- Update the version number and the list of contributors
- Update readme

kcf-jackson commented:
@andycraig Thanks for your work. I have added you as a contributor in the R package, could you please check if the detail is correct in the `DESCRIPTION` file and that you are fine with the new VS Code example in the updated `readme` file? 

By the way, is your repo `languageserver@types` a stable or development release?

andycraig commented:
Thank you for adding me as a contributor! Those details are correct.

The neat thing about language servers is that they will work with any editor that supports the [Language Server Protocol (LSP)](https://microsoft.github.io/language-server-protocol/). VS Code has built-in support for the LSP but there are plugins to provide it for Atom and Sublime and most editors that aren't language-specific. (Emacs and Vim have LSP plugins although I'm not sure how compatible they are with their dedicated R plugins ESS and Nvim-R.) There are instructions for setting up the R languageserver package with different editors [here](https://github.com/REditorSupport/languageserver#language-clients). The `readme` file looks good but it may be worth changing it to 'Integration with editors' and having VS Code as a specific example? Also the type checking will work for unsaved changes in R files as well.

> By the way, is your repo languageserver@types a stable or development release?

My repo is a fork of https://github.com/REditorSupport/languageserver and I reckon it's a good time for me to open an issue about typeChecker support there. Since types in R are experimental at the moment it might be that it stays limited to my fork for the time being though. In that case I'm not quite sure how to think of it in terms of stable or development releases. Happy to hear any suggestions/requests on this point.

kcf-jackson commented:
Thanks for the links on LSP.

> The readme file looks good but it may be worth changing it to 'Integration with editors' and having VS Code as a specific example? Also the type checking will work for unsaved changes in R files as well.

I updated the section heading and removed the phrase about saving the file.

> Since types in R are experimental at the moment it might be that it stays limited to my fork for the time being though. 

Agree. In fact, I just had another round of testing and found that the type-checking breaks when users call a function with the wrong number of arguments. I will try to fix it over the weekend. It is probably also a good time to set up a proper test suite. 

> In that case I'm not quite sure how to think of it in terms of stable or development releases. Happy to hear any suggestions/requests on this point.

Ah, I should be more precise. What I meant to ask is whether major changes are expected in that branch.

andycraig commented:
Thanks for the update to the Readme! No probs re: fixes and tests.

> What I meant to ask is whether major changes are expected in that branch.

I suppose I would periodically merge in changes from the upstream, which could sometimes affect other features (autocompletion, hover help etc.), but I don't anticipate changing the type checking part in any major way. Since it's linked from this repo I'm very happy to handle it cautiously and open an issue here first if I think there's any chance updating the `types` branch might affect type checking. How does that sound?

kcf-jackson commented:
> I suppose I would periodically merge in changes from the upstream, which could sometimes affect other features (autocompletion, hover help etc.), but I don't anticipate changing the type checking part in any major way. Since it's linked from this repo I'm very happy to handle it cautiously and open an issue here first if I think there's any chance updating the types branch might affect type checking. How does that sound?

All good!

I just completed the new test suite, and it can be found at [inst/example/typeChecker_test.R](https://github.com/kcf-jackson/typeChecker/blob/develop/inst/example/typeChecker_test.R). About the wrong-number-of-argument issue, it took me longer than expected to properly handle the case where default values are provided. It is now fixed.

At the moment, when the users provide a default value, the value would automatically imply a type constraint, I wonder if you think this is appropriate? 

If there is no other issue, I'll then proceed to merge the `develop` into the `master` branch. 😎

andycraig commented:
Great! Inferring the type from the default value seems reasonable to me.

No concerns on my side - sounds like it's ready to merge 🎉
